devito/types/basic.py

```diff
@@ -497,8 +497,24 @@ def _new(cls, *args, **kwargs):
                 # Constructor if input is list of list as (row, cols, list_of_list)
                 # doesn't work as it expects a flattened.
                 newobj = super(AbstractTensor, cls)._new(args[2])
+
+            # Filter grid and dimensions
+            grid = {getattr(c, 'grid', None) for c in newobj._mat} - {None}
+            dimensions = {d for c in newobj._mat
+                          for d in getattr(c, 'dimensions', ())} - {None}
+            # If none of the components are devito objects, returns a sympy Matrix
+            if len(grid) == 0 and len(dimensions) == 0:
+                return sympy.ImmutableDenseMatrix(*args)
+            elif len(grid) > 0:
+                dimensions = None
+                grid = grid.pop()
```

Comment:
you probably wanna assert the uniqueness of the grid? ie `assert len(grids) == 1` ?

Reply:
Indeed added